pkg/vault/contracts/authorizer/TimelockAuthorizer.sol

```diff
@@ -101,39 +76,15 @@ contract TimelockAuthorizer is IAuthorizer, TimelockAuthorizerManagement {
         return _isPermissionGranted[actionId][account][where];
     }
 
-    /**
-     * @inheritdoc ITimelockAuthorizer
-     */
-    function hasPermission(
-        bytes32 actionId,
-        address account,
-        address where
-    ) public view override returns (bool) {
+    /// @inheritdoc ITimelockAuthorizer
+    function hasPermission(bytes32 actionId, address account, address where) public view override returns (bool) {
         return _isPermissionGranted[actionId][account][where] || _isPermissionGranted[actionId][account][EVERYWHERE()];
     }
 
     /**
      * @inheritdoc IAuthorizer
      */
-    function canPerform(
-        bytes32 actionId,
-        address account,
-        address where
-    ) public view override returns (bool) {
-        if (msg.sender == address(_authorizerAdaptor)) {
-            // The situation where the caller is the `AuthorizerAdaptor` is a special case, as due to a bug it can be
-            // tricked into passing an incorrect `actionId` value, potentially resulting in escalation of privileges.
-            //
-            // To remedy this we force all calls to the `AuthorizerAdaptor` to be made through a singleton entrypoint
-            // contract, called the `AuthorizerAdaptorEntrypoint`. This contract correctly checks whether `account` can
-            // perform `actionId` on `where`, and then forwards the call onto the `AuthorizerAdaptor` to execute.
-            //
-            // The authorizer then rejects calls to the `AuthorizerAdaptor` which aren't made through the entrypoint,
-            // and approves all calls made through it (since the entrypoint will have already performed any necessary
-            // permission checks).
-            return account == address(_authorizerAdaptorEntrypoint);
-        }
-
+    function canPerform(bytes32 actionId, address account, address where) public view override returns (bool) {
```

Comment:
For reviewers: this is the main change: there's no exception for the adaptor / adaptor entrypoint. Therefore, storing their references is not required anymore.